docs/modules/ROOT/pages/list-of-metrics.adoc

```diff
@@ -94,18 +94,24 @@ in milliseconds).
 
 |status
 |Information regarding the job's status. Potential values:
-0 - submitted;
-1 - initialization phase;
-2 - running;
-3 - suspended;
-4 - exporting snapshot;
-5 - completing phase;
-6 - failed;
-7 - completed successfully;
+
+* 0 - submitted
+* 1 - initialization phase
+* 2 - running
+* 3 - suspended
+* 4 - exporting snapshot
+* 5 - completing phase
+* 6 - failed
+* 7 - completed successfully
 
 |userCancelled
-|Details on whether the user canceled the job. Potential values: 1 - canceled by the user, 0 - otherwise.
+|Details on whether the user canceled the job. Potential values:
+
+* 1 - canceled by the user
+* 0 - otherwise
+
 Exposed via xref:maintain-cluster:monitoring.adoc#via-job-api[Job API] only.
+Note that this metric can be retrieved only if you enable the `storeMetricsAfterJobCompletion` setting for your jobs. See xref:pipelines:configuring-jobs.adoc#storing-metrics[Storing Job Metrics] to learn how to enable it.
```

Comment:
> that this metric can be retrieved only if you enable the `storeMetricsAfterJobCompletion` setting

no, this metrics is available only via `Job.getMetrics()` and the setting is not necessary. But for running jobs userCancelled is always 0.

Reply:
@k-jamroz , I will edit the description for "userCanceled" as you suggested in a previous comment in this PR.

But, here, this was an added content according to the "only for jobs with JobConfig.setStoreMetricsAfterJobCompletion(true)" in https://hazelcast.slack.com/archives/C035HQET5/p1710171595222679. Did I get that wrong?

Reply:
The truth table is as follows - this applies to normal jobs. Metrics for light jobs are partially visible in JMX/Prometheus but that seems to be more by accident than by design. `Job.getMetrics()` is not allowed for light jobs.

| job state | metric | JMX/Prometheus | Job.getMetrics() |
| - | - | - | - |
| running/suspended | userCancelled | not present | present and = 0 |
| running/suspended | other job metrics | present | present |
| finished | userCancelled | not present | present only if storeMetricsAfterJobCompletion |
| finished | other job metrics | not present | present only if storeMetricsAfterJobCompletion |
| failed/cancelled | status and userCancelled | not present | present only if storeMetricsAfterJobCompletion |
| failed/cancelled | other job metrics | not present | not present |

Reply:
we do not return `userCancelled` in JMX because if the job is visible in JMX => it is running => it is not cancelled => metric would be constantly 0.

Reply:
for failed/cancelled jobs we do not keep the runtime metrics regardless of storeMetricsAfterJobCompletion because they could be incomplete/corrupted. But for such jobs status and userCancelled can be reliably returned, so they are the only metrics for such jobs.